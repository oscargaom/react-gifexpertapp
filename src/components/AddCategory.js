import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('')

    const handleChange = (e) => {
        setInputValue(e.target.value);
        //console.log(`handleChange take place`);
    };

    const handleSubmit = (e) => {
        e.preventDefault();

        //console.log(`handleSubmit take place with value: "${inputValue}"`);

        if (inputValue.trim().length > 0 ) {
            setCategories(cat => [inputValue, ...cat]);
            //console.log(`setCategories called with value: "${inputValue}"`);
            setInputValue('');
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type="text"
                value={inputValue}
                onChange={handleChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
