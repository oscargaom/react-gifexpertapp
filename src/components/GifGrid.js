// import React, { useState, useEffect } from 'react'
import React from 'react'
import { useFecthGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';
// import { getGifs } from '../helpers/GetGifs';

export const GifGrid = ({category}) => {

    // const [images, setImages] = useState([]);

    // getGifs();
    const {data: images, loading} = useFecthGifs(category);
    // console.log(data);
    // console.log(loading);


    return (
        <>
            <h3 className="animate__animated animate__fadein">{category}</h3>
            {
                loading && <p className="animate__animated animate__flash">Loading...</p>
            }
            <div className="card-grid">
                {
                    images.map( img => (
                        <GifGridItem 
                            key={img.id} 
                            {...img} 
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}