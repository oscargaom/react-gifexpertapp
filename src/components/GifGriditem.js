import React from 'react'

export const GifGriditem = ({title, url}) => {


    return (
        <div className="card animate__animated animate__fadein">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}
