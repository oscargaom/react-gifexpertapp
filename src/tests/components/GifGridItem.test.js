import React from 'react';
import { GifGridItem } from '../../components/GifGridItem';

const { shallow } = require("enzyme")

describe('Grupo de pruebas de los components', () => {
    const title = 'Titulo';
    const url = 'https://example/img.gif';
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('Prueba componenete <GifGridItem/>', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe tener un parrafo ', () => {
        const parrafo = wrapper.find('p');

        //console.log(parrafo.text().trim());

        expect(parrafo.text().trim()).toBe(title);
    })

    test('Debe tener una imagen con su respectivo src y alt', () => {
        const imagen = wrapper.find('img');

        // console.log(imagen.prop('src'));
        // console.log(imagen.props().alt);

        expect(imagen.prop('src')).toBe(url);
        expect(imagen.props().alt).toBe(title);
    })

    test('Debe tener el estilo animate__fadein', () => {
        const estilo = wrapper.find('div');

        // console.log(estilo.props().className);

        expect(estilo.props().className.includes("animate__fadein")).toBe(true);
    })
    
    
    
    
})
