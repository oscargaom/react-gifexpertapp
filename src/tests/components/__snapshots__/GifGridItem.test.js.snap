// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grupo de pruebas de los components Prueba componenete <GifGridItem/> 1`] = `
<div
  className="card animate__animated animate__fadein"
>
  <img
    alt="Titulo"
    src="https://example/img.gif"
  />
  <p>
    Titulo
  </p>
</div>
`;
